Dart/src/com/jetbrains/lang/dart/ide/runner/server/vmService/DartVmServiceDebugProcess.java

```diff
@@ -279,13 +286,13 @@ public void guessRemoteProjectRoot(@NotNull final ElementList<LibraryRef> librar
 
         LOG.assertTrue(file.getPath().startsWith(localProjectRoot.getPath() + "/"), file.getPath() + "," + localProjectRoot.getPath());
         final String relPath = file.getPath().substring(localProjectRoot.getPath().length()); // starts with slash
-        if (!relPath.startsWith("/lib/") && remoteUri.endsWith(relPath)) {
+        if (relPath.startsWith("/lib/") && remoteUri.endsWith(relPath)) {
           howManyFilesMatch++;
           myRemoteProjectRootUri = remoteUri.substring(0, remoteUri.length() - relPath.length());
         }
       }
 
-      if (howManyFilesMatch == 1) {
+      if (howManyFilesMatch > 0) {
```

Comment:
(I changed this back to the original logic, and added a TODO: to look for additional ways to match if howManyFilesMatch > 1)